index.js

```diff
@@ -1,155 +1,316 @@
-const { existsSync, readFileSync } = require('fs');
-const { join } = require('path');
+/* tslint:disable */
+/* eslint-disable */
+/* prettier-ignore */
 
-const { platform, arch } = process;
+/* auto-generated by NAPI-RS */
 
-let nativeBinding = null;
-let localFileExisted = false;
-let loadError = null;
+const { existsSync, readFileSync } = require('fs')
+const { join } = require('path')
+
+const { platform, arch } = process
+
+let nativeBinding = null
+let localFileExisted = false
+let loadError = null
 
 function isMusl() {
   // For Node 10
   if (!process.report || typeof process.report.getReport !== 'function') {
     try {
-      return readFileSync('/usr/bin/ldd', 'utf8').includes('musl');
+      const lddPath = require('child_process').execSync('which ldd').toString().trim()
+      return readFileSync(lddPath, 'utf8').includes('musl')
     } catch (e) {
-      return true;
+      return true
     }
   } else {
-    const { glibcVersionRuntime } = process.report.getReport().header;
-    return !glibcVersionRuntime;
+    const { glibcVersionRuntime } = process.report.getReport().header
+    return !glibcVersionRuntime
   }
 }
 
 switch (platform) {
+  case 'android':
```

Comment:
What does adding `'android'` and `'ia32'` cases do and what inspired you to make this change?

Reply:
This whole file is auto-generated by our tooling (see the comment at the top of the file).  It seems like this file changed a bit when I upgraded the CLI, so I decided to regenerate it and check it in.

Reply:
It should be okay then, thanks for clarifying! 

From my quick google search, seems like no semicolons aren't that unstandard in JS, but I'll leave this thread open for team review in case anyone else wants to chime in.

Reply:
Despite being auto generated I recall we did remove the unsupported platforms intentionally. I think we probably should've removed that comment, I think this file could be considered manually maintained now.

Reply:
Okay, so for now I have:

1. added the napi-cli to the exclude list in the dependabot so we don't try to update automatically in the future
2. I manually removed the two unsupported platforms and left this dependency upgrade

Let me know if that sounds good to everyone.

Reply:
> I manually removed the two unsupported platforms

Which two? I'm seeing more that I don't recognize, freebsd, arm64 windows, riscv64 linux, etc. Shouldn't we be able to revert this 🤔 I am not sure what we're bringing in

While the changes seem generally correct and good I worry about changes like:

```diff
- return readFileSync('/usr/bin/ldd', 'utf8').includes('musl');
+ const lddPath = require('child_process').execSync('which ldd').toString().trim()
+ return readFileSync(lddPath, 'utf8').includes('musl');
```

Turning a file read into a process launch and file read not being surfaced appropriately so folks aren't surprised by a compression library launching a subprocess.

Reply:
To clarify we should revert any changes to `index.js`, and we can update this in a follow-up (we can decide on priority in triage).